src/layers/misc/channelwise_softmax.cu

```diff
@@ -438,21 +445,28 @@ void bp_impl(size_t num_channels,
   // dot(y,dL/dy)
   LocalMat local_y_dot_dy(num_channels, local_mini_batch_size);
   El::Zero(local_y_dot_dy);
+
+  auto multisync = El::MakeMultiSync(gpu::get_sync_info(local_input_grad),
+                                     gpu::get_sync_info(local_output_grad),
+                                     gpu::get_sync_info(local_output),
+                                     gpu::get_sync_info(local_y_dot_dy));
```

Comment:
In this case, `local_y_dot_dy` must come first, because `local_input_grad` doesn't participate in the first kernel below. In a futuristic multistream world, this could cause issues. Of course, `local_y_dot_dy` could, in that universe, be setup to have the right syncinfo so this could never happen.
```suggestion
  auto multisync = El::MakeMultiSync(gpu::get_sync_info(local_y_dot_dy),
                                     gpu::get_sync_info(local_output_grad),
                                     gpu::get_sync_info(local_output),
                                     gpu::get_sync_info(local_input_grad));
```